### ISSUE TITLE
Could not find function "RunMiQC"

### ISSUE THREAD

**Issue by denvercal1234GitHub** [satijalab/seurat#5825]
Hi there, 

The function "RunMiQC" is no longer available. Would you mind informing me how to proceed?

Ref: http://htmlpreview.github.io/?https://github.com/satijalab/seurat-wrappers/blob/master/docs/miQC.html

Thank you for your help.

**Comment by samuel-marsh** [comment]
Hi,

Not member of dev team but hopefully can be helpful.  Can you provide full code you are running and the output of sessionInfo?

thanks!
Sam

**Comment by denvercal1234GitHub** [comment]
Hi @samuel-marsh - Thanks for the response. 

Below is the code and sessioninfo. 

Also, if I use miQC, should I not set min.cell right?

Thanks again for your help!

```
#flexmix which is wrapped by the miQC package
install.packages("flexmix")

remotes::install_github("greenelab/miQC", build_vignettes = F)

library(miQC)
library(Seurat)
library(SeuratData)
library(SeuratWrappers)
library(flexmix)

obj.data <- Read10X(data.dir = "~outs/count/filtered_feature_bc_matrix")

obj <- CreateSeuratObject(counts = objdata, names.delim = "-", names.field = 2)

obj <- PercentageFeatureSet(obj, pattern = "^MT-", col.name="percent.mt")

obj<- RunMiQC(obj, percent.mt = "percent.mt", nFeature_RNA = "nFeature_RNA", posterior.cutoff = 0.75, model.slot = "flexmix_model")
```


```
sessionInfo()
```

R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: OS X  12.3.1

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] pbmc3k.SeuratData_3.1.4   panc8.SeuratData_3.0.2    ifnb.SeuratData_3.1.0     hcabm40k.SeuratData_3.0.0
 [5] SeuratData_0.2.1          SeuratWrappers_0.3.0      flexmix_2.3-17            lattice_0.20-45          
 [9] ggmin_0.0.0.9000          ggpubr_0.4.0.999          pheatmap_1.0.12           cowplot_1.1.1            
[13] ggrepel_0.9.1             purrr_0.3.4               metap_1.8                 multtest_2.46.0          
[17] Biobase_2.50.0            BiocGenerics_0.36.1       BiocFileCache_1.14.0      dbplyr_2.1.1             
[21] Nebulosa_1.0.2            patchwork_1.1.1           ggplot2_3.3.5.9000        limma_3.46.0             
[25] pastecs_1.3.21            gdata_2.18.0              tidyr_1.2.0               Matrix_1.4-1             
[29] dplyr_1.0.8               readr_2.1.2               SeuratObject_4.0.4        Seurat_4.1.0.9003        

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3              scattermore_0.8             bit64_4.0.5                
  [4] knitr_1.38                  irlba_2.3.5                 multcomp_1.4-18            
  [7] DelayedArray_0.16.3         data.table_1.14.2           rpart_4.1.16               
 [10] RCurl_1.98-1.6              generics_0.1.2              callr_3.7.0                
 [13] TH.data_1.1-0               usethis_2.1.5               RSQLite_2.2.12             
 [16] RANN_2.6.1                  future_1.24.0               bit_4.0.4                  
 [19] tzdb_0.3.0                  mutoss_0.1-12               spatstat.data_2.1-4        
 [22] httpuv_1.6.5                SummarizedExperiment_1.20.0 assertthat_0.2.1           
 [25] viridis_0.6.2               xfun_0.30                   hms_1.1.1                  
 [28] promises_1.2.0.1            fansi_1.0.3                 igraph_1.3.0               
 [31] DBI_1.1.2                   tmvnsim_1.0-2               htmlwidgets_1.5.4          
 [34] spatstat.geom_2.4-0         stats4_4.0.2                ellipsis_0.3.2             
 [37] ks_1.13.4                   backports_1.4.1             deldir_1.0-6               
 [40] sparseMatrixStats_1.2.1     MatrixGenerics_1.2.1        vctrs_0.4.0                
 [43] SingleCellExperiment_1.12.0 remotes_2.4.2               here_1.0.1                 
 [46] ROCR_1.0-11                 abind_1.4-5                 cachem_1.0.6               
 [49] withr_2.5.0                 sctransform_0.3.3           prettyunits_1.1.1          
 [52] mclust_5.4.9                goftest_1.2-3               mnormt_2.0.2               
 [55] cluster_2.1.3               lazyeval_0.2.2              crayon_1.5.1               
 [58] labeling_0.4.2              edgeR_3.32.1                pkgconfig_2.0.3            
 [61] qqconf_1.2.3                GenomeInfoDb_1.26.7         nlme_3.1-157               
 [64] vipor_0.4.5                 pkgload_1.2.4               nnet_7.3-17                
 [67] devtools_2.4.3              rlang_1.0.2                 globals_0.14.0             
 [70] lifecycle_1.0.1             miniUI_0.1.1.1              sandwich_3.0-1             
 [73] mathjaxr_1.6-0              rsvd_1.0.5                  rprojroot_2.0.3            
 [76] polyclip_1.10-0             matrixStats_0.61.0          lmtest_0.9-40              
 [79] carData_3.0-5               boot_1.3-28                 zoo_1.8-9                  
 [82] beeswarm_0.4.0              ggridges_0.5.3              processx_3.5.3             
 [85] png_0.1-7                   viridisLite_0.4.0           bitops_1.0-7               
 [88] KernSmooth_2.23-20          blob_1.2.2                  DelayedMatrixStats_1.12.3  
 [91] stringr_1.4.0               parallelly_1.30.0           spatstat.random_2.2-0      
 [94] rstatix_0.7.0               S4Vectors_0.28.1            ggsignif_0.6.3             
 [97] beachmat_2.6.4              scales_1.1.1                memoise_2.0.1              
[100] magrittr_2.0.3              plyr_1.8.7                  ica_1.0-2                  
[103] zlibbioc_1.36.0             compiler_4.0.2              RColorBrewer_1.1-3         
[106] plotrix_3.8-2               fitdistrplus_1.1-8          cli_3.2.0                  
[109] XVector_0.30.0              listenv_0.8.0               pbapply_1.5-0              
[112] ps_1.6.0                    MASS_7.3-56                 mgcv_1.8-40                
[115] tidyselect_1.1.2            stringi_1.7.6               BiocSingular_1.6.0         
[118] locfit_1.5-9.4              grid_4.0.2                  tools_4.0.2                
[121] future.apply_1.8.1          rstudioapi_0.13             gridExtra_2.3              
[124] farver_2.1.0                Rtsne_0.15                  digest_0.6.29              
[127] BiocManager_1.30.16         shiny_1.7.1                 pracma_2.3.8               
[130] Rcpp_1.0.8.3                GenomicRanges_1.42.0        car_3.0-12                 
[133] broom_0.7.12                scuttle_1.0.4               later_1.3.0                
[136] RcppAnnoy_0.0.19            httr_1.4.2                  Rdpack_2.3                 
[139] colorspace_2.0-3            brio_1.1.3                  fs_1.5.2                   
[142] tensor_1.5                  reticulate_1.24             IRanges_2.24.1             
[145] splines_4.0.2               uwot_0.1.11                 sn_2.0.2                   
[148] spatstat.utils_2.3-0        scater_1.18.6               plotly_4.10.0              
[151] sessioninfo_1.2.2           xtable_1.8-4                jsonlite_1.8.0             
[154] modeltools_0.2-23           testthat_3.1.3              R6_2.5.1                   
[157] TFisher_0.2.0               pillar_1.7.0                htmltools_0.5.2            
[160] mime_0.12                   glue_1.6.2                  fastmap_1.1.0              
[163] BiocParallel_1.24.1         BiocNeighbors_1.8.2         codetools_0.2-18           
[166] pkgbuild_1.3.1              mvtnorm_1.1-3               utf8_1.2.2                 
[169] spatstat.sparse_2.1-0       tibble_3.1.6                numDeriv_2016.8-1.1        
[172] curl_4.3.2                  ggbeeswarm_0.6.0            leiden_0.3.9               
[175] gtools_3.9.2                survival_3.3-1              desc_1.4.1                 
[178] munsell_0.5.0               GenomeInfoDbData_1.2.4      reshape2_1.4.4             
[181] gtable_0.3.0                rbibutils_2.2.7             spatstat.core_2.4-2

**Comment by denvercal1234GitHub** [comment]
Hi @samuel-marsh - Thanks for the response. 

I restarted R, and my laptop. Reinstalled the package and activated it (library()), but still had the same issue.

<img width="838" alt="Screen Shot 2022-05-16 at 12 35 17 PM" src="https://user-images.githubusercontent.com/62179082/168584113-674288d3-f1f7-4714-986b-e8e5e1da724a.png">

**Comment by denvercal1234GitHub** [comment]
<img width="814" alt="Screen Shot 2022-05-16 at 12 36 56 PM" src="https://user-images.githubusercontent.com/62179082/168584345-77585449-4ec2-4c25-8521-57ccb14ff2e6.png">

**Comment by denvercal1234GitHub** [comment]
miQC package is not available for R version 4.0.2?

```
> remove.packages("miQC")
Removing package from ‘/Library/Frameworks/R.framework/Versions/4.0/Resources/library’
(as ‘lib’ is unspecified)
> if (!requireNamespace("BiocManager"))
+     install.packages("BiocManager")
> BiocManager::install("miQC")
'getOption("repos")' replaces Bioconductor standard repositories, see '?repositories' for details

replacement repositories:
    CRAN: https://cran.rstudio.com/

Bioconductor version 3.12 (BiocManager 1.30.17), R 4.0.2 (2020-06-22)
Installing package(s) 'miQC'
Warning: package ‘miQC’ is not available (for R version 4.0.2)
Old packages: 'exactextractr', 'nloptr', 'pdftools', 'rgdal'
Update all/some/none? [a/s/n]: 
a

  There are binary versions available but the source versions are later:

Do you want to install from sources the packages which need compilation? (Yes/no/cancel) Yes
installing the source packages ‘exactextractr’, ‘nloptr’, ‘pdftools’, ‘rgdal’

trying URL 'https://cran.rstudio.com/src/contrib/exactextractr_0.8.2.tar.gz'
Content type 'application/x-gzip' length 623313 bytes (608 KB)
==================================================
downloaded 608 KB

trying URL 'https://cran.rstudio.com/src/contrib/nloptr_2.0.1.tar.gz'
Content type 'application/x-gzip' length 2219799 bytes (2.1 MB)
==================================================
downloaded 2.1 MB

trying URL 'https://cran.rstudio.com/src/contrib/pdftools_3.2.0.tar.gz'
Content type 'application/x-gzip' length 935874 bytes (913 KB)
==================================================
downloaded 913 KB

trying URL 'https://cran.rstudio.com/src/contrib/rgdal_1.5-32.tar.gz'
Content type 'application/x-gzip' length 4469369 bytes (4.3 MB)
==================================================
downloaded 4.3 MB

* installing *source* package ‘exactextractr’ ...
** package ‘exactextractr’ successfully unpacked and MD5 sums checked
** using staged installation
configure: exactextractr: 0.8.2
checking for geos-config... no
configure: error: geos-config not found or not executable
ERROR: configuration failed for package ‘exactextractr’
* removing ‘/Library/Frameworks/R.framework/Versions/4.0/Resources/library/exactextractr’
* restoring previous ‘/Library/Frameworks/R.framework/Versions/4.0/Resources/library/exactextractr’
Warning in install.packages(update[instlib == l, "Package"], l, repos = repos,  :
  installation of package ‘exactextractr’ had non-zero exit status
* installing *source* package ‘nloptr’ ...
** package ‘nloptr’ successfully unpacked and MD5 sums checked
** using staged installation
checking whether the C++ compiler works... yes
checking for C++ compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C++... yes
checking whether clang++ -mmacosx-version-min=10.13 -std=gnu++11 accepts -g... yes
checking for clang++ -mmacosx-version-min=10.13 -std=gnu++11 option to enable C++11 features... none needed
checking how to run the C++ preprocessor... clang++ -mmacosx-version-min=10.13 -std=gnu++11 -E
checking whether the compiler supports GNU C++... (cached) yes
checking whether clang++ -mmacosx-version-min=10.13 -std=gnu++11 accepts -g... (cached) yes
checking for clang++ -mmacosx-version-min=10.13 -std=gnu++11 option to enable C++11 features... (cached) none needed
checking for pkg-config... /usr/local/bin/pkg-config
checking if pkg-config knows NLopt... no
checking for cmake... no

------------------ CMAKE NOT FOUND --------------------

CMake was not found on the PATH. Please install CMake:

 - sudo yum install cmake          (Fedora/CentOS; inside a terminal)
 - sudo apt install cmake          (Debian/Ubuntu; inside a terminal).
 - sudo pacman -S cmake            (Arch Linux; inside a terminal).
 - sudo brew install cmake         (MacOS; inside a terminal with Homebrew)
 - sudo port install cmake         (MacOS; inside a terminal with MacPorts)

Alternatively install CMake from: <https://cmake.org/>

-------------------------------------------------------

ERROR: configuration failed for package ‘nloptr’
* removing ‘/Library/Frameworks/R.framework/Versions/4.0/Resources/library/nloptr’
* restoring previous ‘/Library/Frameworks/R.framework/Versions/4.0/Resources/library/nloptr’
Warning in install.packages(update[instlib == l, "Package"], l, repos = repos,  :
  installation of package ‘nloptr’ had non-zero exit status
* installing *source* package ‘pdftools’ ...
** package ‘pdftools’ successfully unpacked and MD5 sums checked
** using staged installation
Package poppler-cpp was not found in the pkg-config search path.
Perhaps you should add the directory containing `poppler-cpp.pc'
to the PKG_CONFIG_PATH environment variable
No package 'poppler-cpp' found
Homebrew 3.4.4
Homebrew/homebrew-core (git revision 0c65c44d487; last commit 2022-04-01)
Homebrew/homebrew-cask (git revision 31060c3207; last commit 2022-04-01)
Using PKG_CFLAGS=-I/usr/local/opt/poppler/include/cpp
Using PKG_LIBS=-L/usr/local/opt/poppler/lib -lpoppler-cpp
--------------------------- [ANTICONF] --------------------------------
Configuration failed to find 'poppler-cpp' system library. Try installing:
 * rpm: poppler-cpp-devel (Fedora, CentOS, RHEL)
 * brew: poppler (MacOS)
 * deb: libpoppler-cpp-dev (Debian, Ubuntu, etc)
 * On Ubuntu 16.04 or 18.04 use this PPA:
    sudo add-apt-repository -y ppa:cran/poppler
    sudo apt-get update
    sudo apt-get install -y libpoppler-cpp-dev
If poppler-cpp is already installed, check that 'pkg-config' is in your
PATH and PKG_CONFIG_PATH contains a poppler-cpp.pc file. If pkg-config
is unavailable you can set INCLUDE_DIR and LIB_DIR manually via:
R CMD INSTALL --configure-vars='INCLUDE_DIR=... LIB_DIR=...'
-------------------------- [ERROR MESSAGE] ---------------------------
<stdin>:1:10: fatal error: 'poppler-document.h' file not found
#include <poppler-document.h>
         ^~~~~~~~~~~~~~~~~~~~
1 error generated.
--------------------------------------------------------------------
ERROR: configuration failed for package ‘pdftools’
* removing ‘/Library/Frameworks/R.framework/Versions/4.0/Resources/library/pdftools’
* restoring previous ‘/Library/Frameworks/R.framework/Versions/4.0/Resources/library/pdftools’
Warning in install.packages(update[instlib == l, "Package"], l, repos = repos,  :
  installation of package ‘pdftools’ had non-zero exit status
* installing *source* package ‘rgdal’ ...
** package ‘rgdal’ successfully unpacked and MD5 sums checked
** using staged installation
configure: R_HOME: /Library/Frameworks/R.framework/Resources
configure: CC: clang -mmacosx-version-min=10.13
configure: CXX: clang++ -mmacosx-version-min=10.13 -std=gnu++11
configure: CFLAGS: -Wall -g -O2
configure: CPPFLAGS: -I/usr/local/include
configure: CXXFLAGS: -Wall -g -O2
configure: LDFLAGS: -L/usr/local/lib
configure: LDFLAGS: -L/usr/local/lib
configure: CXX11 is: clang++ -mmacosx-version-min=10.13, CXX11STD is: -std=gnu++11
configure: CXX is: clang++ -mmacosx-version-min=10.13 -std=gnu++11
configure: C++11 support available
configure: rgdal: 1.5-32
checking for /usr/bin/svnversion... no
configure: svn revision: 1176
checking for gdal-config... no
no
configure: error: gdal-config not found or not executable.
ERROR: configuration failed for package ‘rgdal’
* removing ‘/Library/Frameworks/R.framework/Versions/4.0/Resources/library/rgdal’
* restoring previous ‘/Library/Frameworks/R.framework/Versions/4.0/Resources/library/rgdal’
Warning in install.packages(update[instlib == l, "Package"], l, repos = repos,  :
  installation of package ‘rgdal’ had non-zero exit status

The downloaded source packages are in
	‘/private/var/folders/yf/mfkcsd412qndhd65fdwcz7l40000gn/T/RtmpXUcKEz/downloaded_packages’
```

**Comment by denvercal1234GitHub** [comment]
It worked now after I reinstalled SeuratWrappers and flexmix packages. 😅. Thank you all
